### PR TITLE
enable increment build for build_libtorch

### PR DESCRIPTION
**Pull request by mszhanyi**
Since issue #59859 is resolved.

rerun_cmake in build_libtorch should not be hardcoded.
build_libtorch is necessary to generate debug version libtorch.

### CODE REVIEW

**Review by peterjc123 — COMMENTED** [review]
LG, but I don't know if there is any script that relies on the `rerun_cmake` option here. (you switched the default value to False). It would be best that you could open a PR to test this change.

**Comment by mszhanyi** [comment]
> LG, but I don't know if there is any script that relies on the `rerun_cmake` option here. (you switched the default value to False). It would be best that you could open a PR to test this change.

@peterjc123 , It is obvious that no other script relies on the option because it is under `if __name__ == '__main__'`. The old logic is not reasonable because `python ..\tools\build_libtorch.py ` reruned cmake every time. Now it is consistent with `python setup.py   install`

**Comment by peterjc123** [comment]
@mszhanyi I remember that the Linux binary jobs used to build the shared and static variant of libtorch in a row using a for-loop in the bash script in the past. Under this case, setting rerun_cmake=False may be risky. But I'm unsure if they are doing it now. Will need a double check.

**Comment by mszhanyi** [comment]
> @mszhanyi I remember that the Linux binary jobs used to build the shared and static variant of libtorch in a row using a for-loop in the bash script in the past. Under this case, setting rerun_cmake=False may be risky. But I'm unsure if they are doing it now. Will need a double check.

@peterjc123, I run all workflows. It should be OK. The failed workflow has nothing with this change. #63139

**Review by peterjc123 — APPROVED** [review]

**Review by seemethere — APPROVED** [review]

File: tools/build_libtorch.py [hunk 1]
```diff
@@ -14,7 +14,10 @@
 if __name__ == '__main__':
     # Placeholder for future interface. For now just gives a nice -h.
     parser = argparse.ArgumentParser(description='Build libtorch')
+    parser.add_argument('--cmake', action="store_true", help='rerun cmake')
```
Comment (malfet) [inline tools/build_libtorch.py:17]: Can you please rename it to `--rerun-cmake`?

**Review by malfet — APPROVED** [review]
LGTM, but please consider renaming the option to `--rerun-cmake` (I've checked, that if `build_caffe2` is called with rerun_cmake set to False, it will still call it once.)

File: tools/build_libtorch.py (hunk 1 above)
Comment (malfet) [inline tools/build_libtorch.py:17]:
```suggestion
    parser.add_argument('--rerun-cmake', action="store_true", help='rerun cmake')
```

File: tools/build_libtorch.py [hunk 2]
```diff
@@ -14,7 +14,10 @@
...
+    parser.add_argument('--cmake_only', action="store_true",
```
Comment (malfet) [inline tools/build_libtorch.py:18]:
```suggestion
    parser.add_argument('--cmake-only', action="store_true",
```

File: tools/build_libtorch.py [hunk 3]
```diff
@@ -14,7 +14,10 @@
...
+                        help='Stop once cmake terminates. Leave users a chance to adjust build options')
     options = parser.parse_args()
 
     build_caffe2(version=None, cmake_python_library=None, build_python=False,
-                 rerun_cmake=True, cmake_only=False, cmake=CMake())
+                 rerun_cmake=options.cmake, cmake_only=options.cmake_only, cmake=CMake())
```
Comment (malfet) [inline tools/build_libtorch.py:23]:
```suggestion
                 rerun_cmake=options.rerun_cmake, cmake_only=options.cmake_only, cmake=CMake())
```

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**